Lib/threading.py

```diff
@@ -1451,11 +1451,12 @@ def _stop(self):
         pass
 
     def is_alive(self):
-        assert not self._is_stopped and self._started.is_set()
-        return True
+        if not self._is_stopped and self._started.is_set():
+            return True
+        raise AssertionError("thread is not alive")
```

Comment:
I am not sure that any change is needed here.  If so, I think another error.  See review comment.

Reply:
I think it is needed. `_DummyThread` is assumed to always be alive, but as I showed in the test case, it is easy to kill it (sort of). So, adding an error about reaching an unwanted state seems like a good thing to do.

I agree that `AssertionError` is probably not the best way to raise an error, will change it to `RuntimeError`.

Reply:
Again, the counter example would be:

```python
import threading, _thread

def f(mutex):
    threading.current_thread()
    mutex.release()

mutex = threading.Lock()
mutex.acquire()

tid = _thread.start_new_thread(f, (mutex,))
mutex.acquire()

threading._active[tid]._started.clear()
threading._active[tid].is_alive()
```

With asserts on `main` it will raise `AssertionError`
With `-OO` on `main` it will return `True`

I think that this is not fine.